### PR TITLE
Validate storage pools in CreateVolume

### PR DESCRIPTION
**Pull request by amacaskill**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test

/kind feature

> /kind flake

**What this PR does / why we need it**:
This PR validates storage pools passed to the CreateVolume request in parameters. This PR contains 2 commits: 
1. Adds storage pool helper functions to common package, and change DiskParameters to use StoragePool type instead of a string.
2. Adds the validation for storage pools in CreateVolume. 

A follow up PR will add enable_storage_pool to the already existing csidriver_operation_errors metric. 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Adds validation and support for creating volumes in storage pools.
```

### CODE REVIEW

File: pkg/common/parameters_test.go [hunk 1]
```diff
@@ -199,6 +201,37 @@ func TestExtractAndDefaultParameters(t *testing.T) {
...
+					},
+				},
+			},
+		},
+		{
+			name:       "invalid storage pool parameters",
+			parameters: map[string]string{ParameterKeyType: "hyperdisk-balanced", ParameterKeyStoragePools: "zones/us-central1-a/storagePools/storagePool-1"},
+			labels:     map[string]string{},
+			expectErr:  true,
+		},
```
Comment (mattcary) [inline pkg/common/parameters_test.go:234]: Add a test that tries to break the regular expression, eg /zones/zones and stuff like that.

**Comment by mattcary** [comment]
Just the one nit then it will lgtm

File: pkg/common/parameters_test.go (hunk 1 above)
Comment (amacaskill) [reply pkg/common/parameters_test.go:240]: Added.

**Comment by amacaskill** [comment]
> Just the one nit then it will lgtm

Done!

**Pull request merged**